src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -61,42 +61,30 @@ public static bool Equals(ReadOnlySpan<char> left, ReadOnlySpan<char> right)
                     }
                 }
             }
-            else if (Avx.IsSupported && length >= (uint)Vector256<TRight>.Count)
+            else if (Avx.IsSupported && length >= (uint)Vector256<TLeft>.Count)
             {
                 ref TLeft currentLeftSearchSpace = ref left;
-                ref TLeft oneVectorAwayFromLeftEnd = ref Unsafe.Add(ref currentLeftSearchSpace, length - TLoader.Count256);
                 ref TRight currentRightSearchSpace = ref right;
-                ref TRight oneVectorAwayFromRightEnd = ref Unsafe.Add(ref currentRightSearchSpace, length - (uint)Vector256<TRight>.Count);
-
-                Vector256<TRight> leftValues;
-                Vector256<TRight> rightValues;
+                ref TRight oneVectorAwayFromRightEnd = ref Unsafe.Add(ref currentRightSearchSpace, length - (uint)Vector256<TLeft>.Count);
```

Comment:
I'm not understanding why this is valid.  We're subtracting from the "right" search space the number of "left" elements in a vector?

Reply:
It works because `TLeft` and `TRight` are either the same type, or we are in the widen case where `Vector<TLeft>` is twice the size of `Vector<TRight>` and the widen code will advance twice `Vector<TRight>.Count` which is equal to 1 `Vector<TLeft>.Count`.

But it is written in a confusing way. The whole `TLoader` abstraction helps with code sharing but makes this part kind of yucky. Maybe if the compare method advanced the pointers it would be better?

Reply:
Thanks. At a minimum a comment explaining would be helpful.